cliext/client.go

```diff
@@ -0,0 +1,288 @@
+		profile.APIKey = cfg.ApiKey
+	}
+
+	// Handle gRPC metadata from flags.
+	if len(cfg.GrpcMeta) > 0 {
+		grpcMetaFromArg, err := parseKeyValuePairs(cfg.GrpcMeta)
+		if err != nil {
+			return client.Options{}, fmt.Errorf("invalid gRPC meta: %w", err)
+		}
+		if len(profile.GRPCMeta) == 0 {
+			profile.GRPCMeta = make(map[string]string, len(cfg.GrpcMeta))
+		}
+		for k, v := range grpcMetaFromArg {
+			profile.GRPCMeta[k] = v
+		}
+	}
+
+	// If any of these TLS values are present, set TLS if not set, and set values.
+	// NOTE: This means that tls=false does not explicitly disable TLS when set
+	// via envconfig.
+	if cfg.Tls ||
+		cfg.TlsCertPath != "" || cfg.TlsKeyPath != "" || cfg.TlsCaPath != "" ||
+		cfg.TlsCertData != "" || cfg.TlsKeyData != "" || cfg.TlsCaData != "" {
+		if profile.TLS == nil {
+			profile.TLS = &envconfig.ClientConfigTLS{}
+		}
+		if cfg.TlsCertPath != "" {
+			profile.TLS.ClientCertPath = cfg.TlsCertPath
+		}
+		if cfg.TlsCertData != "" {
+			profile.TLS.ClientCertData = []byte(cfg.TlsCertData)
+		}
+		if cfg.TlsKeyPath != "" {
+			profile.TLS.ClientKeyPath = cfg.TlsKeyPath
+		}
+		if cfg.TlsKeyData != "" {
+			profile.TLS.ClientKeyData = []byte(cfg.TlsKeyData)
+		}
+		if cfg.TlsCaPath != "" {
+			profile.TLS.ServerCACertPath = cfg.TlsCaPath
+		}
+		if cfg.TlsCaData != "" {
+			profile.TLS.ServerCACertData = []byte(cfg.TlsCaData)
+		}
+		if cfg.TlsServerName != "" {
+			profile.TLS.ServerName = cfg.TlsServerName
+		}
+		if cfg.TlsDisableHostVerification {
+			profile.TLS.DisableHostVerification = cfg.TlsDisableHostVerification
+		}
+	}
+
+	// If TLS is explicitly disabled (--tls=false), we turn it off. Otherwise it may be
+	// implicitly enabled if API key or any other TLS setting is set.
+	if cfg.FlagSet != nil && cfg.FlagSet.Changed("tls") && !cfg.Tls {
+		profile.TLS = &envconfig.ClientConfigTLS{Disabled: true}
+	}
+
+	// If codec endpoint is set, create codec setting regardless. But if auth is
+	// set, it only overrides if codec is present.
+	if cfg.CodecEndpoint != "" {
+		if profile.Codec == nil {
+			profile.Codec = &envconfig.ClientConfigCodec{}
+		}
+		profile.Codec.Endpoint = cfg.CodecEndpoint
+	}
+	if cfg.CodecAuth != "" && profile.Codec != nil {
+		profile.Codec.Auth = cfg.CodecAuth
+	}
+
+	// Convert profile to client options.
+	clientOpts, err := profile.ToClientOptions(envconfig.ToClientOptionsRequest{})
+	if err != nil {
+		return client.Options{}, fmt.Errorf("failed to build client options: %w", err)
+	}
+
+	// Set client authority if provided.
+	if cfg.ClientAuthority != "" {
+		clientOpts.ConnectionOptions.Authority = cfg.ClientAuthority
+	}
+
+	// Set identity if provided.
+	if cfg.Identity != "" {
+		clientOpts.Identity = cfg.Identity
+	}
+
+	// Set logger if provided.
+	if b.Logger != nil {
+		clientOpts.Logger = log.NewStructuredLogger(b.Logger)
+	}
+
+	// Attempt to configure OAuth config if no API key is set.
+	if cfg.ApiKey == "" {
+		result, err := LoadClientOAuth(LoadClientOAuthOptions{
```

Comment:
I don't think this properly accounts for `if !common.DisableConfigFile || !common.DisableConfigEnv` that the other envconfig loading does above.

Would be ok if we wanted to go ahead and have Go SDK's env config support "additional data" to avoid this double-load for everyone that is doing basic no-auth CLI calls (happens commonly in cases like proxies). But also ok if we want to double load while we wait for that.

Reply:
👍 Absolutely; adding that check now. (just for `DisableConfigFile` since the OAuth cannot be configured via env vars)